src/mpid/ch4/src/ch4_progress.h

```diff
@@ -34,8 +34,9 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_Progress_test(int flags)
     }
 #endif
 
-    if (OPA_load_int(&MPIDI_CH4_Global.registered_progress_hooks) && (flags & MPIDI_PROGRESS_HOOKS)) {
-        for (i = 0; i < MAX_PROGRESS_HOOKS; i++) {
+    if (num_of_hooks = OPA_load_int(&MPIDI_CH4_Global.registered_progress_hooks) &&
```

Comment:
Might need a double parenthesis to avoid compiler warnings.

Reply:
This is no longer needed, if the above mentioned change is made.

Reply:
I put the assignment into a different statement in the commit.